algorithms/ar-binsrt/typescript/bin_sort.ts

```diff
@@ -0,0 +1,32 @@
+function binSort(collection: Array<number>, size:number): void {
+    if(size <= 0) return;
+
+    var arrayList: number[][] = new Array<number>(size).map(() => new Array());
+
+    for(let i=0; i<size; i++){
+        arrayList[i] = new Array()
+    }
+
+    for(let i=0; i<size; i++){
+        var index = Math.floor(collection[i]) * size
+        arrayList[index].push(collection[i])
+    }
+
+    for(let element of arrayList){
+        element.sort()
+    }
+
+    var index: number = 0
+    for(let i=0; i<size; i++){
+        for(let j=0, arraySize=arrayList[i].length; j<arraySize; j++){
+            collection[index++] = arrayList[i][j]
```

Comment:
Codacy found an issue: ['index' used outside of binding context.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6395193)